server/datastore/mysql/migrations/tables/20220818101352_ChangeSoftwareVendorWidth.go

```diff
@@ -0,0 +1,78 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220818101352, Down_20220818101352)
+}
+
+func Up_20220818101352(tx *sql.Tx) error {
+	logger.Info.Println("Increasing width of software.vendor...")
+
+	//-----------------
+	// Add temp column.
+	//-----------------
+	if _, err := tx.Exec(
+		`ALTER TABLE software ADD COLUMN vendor_wide varchar(114) NULL, ALGORITHM=INPLACE, LOCK=NONE`); err != nil {
+		return errors.Wrapf(err, "creating temp column for vendor")
+	}
+
+	//---------------------
+	// Add uniq constraint
+	//---------------------
+	if _, err := tx.Exec(
+		"ALTER TABLE software ADD constraint unq_name UNIQUE (name, version, source, `release`, vendor_wide, arch)"); err != nil {
+		return errors.Wrapf(err, "adding new uniquess constraint")
+	}
+
+	//------------------
+	// Update in batches
+	//------------------
+	const updateStmt = `UPDATE software SET vendor_wide = vendor WHERE vendor_wide IS NULL LIMIT 500`
```

Comment:
`... AND vendor NOT NULL`? Not sure if that'll make the update faster really.

Reply:
Unfortunately we don't have an index on `vendor` - so that won't help much